### DISCUSSION TITLE
Teleport dynamic forwarding constantly loses connection

### DISCUSSION THREAD

**Discussion by techcarlin** [gravitational/teleport#9943] [Q&A]
Hello there. 
I spun up Teleport on a Ubuntu VM the other day and got it setup. All seems to work just fine via the browser etc, MFA and ssh into the server seems top notch all the time. 

However, when I try to initiate a SOCKS proxy through dynamic port forwarding on my ubuntu linux client through TSH I get this all the time - randomly disconnects and causes Firefox to lose connection to the proxy server. Any thoughts?

$ **tsh ssh -D 127.0.0.1:9998 tech@ubuntuserver**
tech@ubuntuserver:~$ logout
the connection was closed on the remote side on  25 Jan 22 14:45 MST
tech@tech-XPS-13-7390:~$ **tsh ssh -D 9998 tech@ubuntuserver**
tech@ubuntuserver:~$ ERROR: wait: remote command exited without exit status or exit signal

tech@tech-XPS-13-7390:~$ 

I've tried it both ways for the command and both end up in this same error in a few short minutes.... (10)? Sometimes sooner, all random.

Thanks for any insight. I have port 443, 3023 to 3025, and 3080 forwarded to the Teleport box.

**Comment by techcarlin** [comment]
![teleport](https://user-images.githubusercontent.com/98430194/151067708-75e38574-952a-4a19-a9de-32b0fe8c7ab3.png)
Sorry, Maybe this picture helps describe the issue. 

> Hello there. I spun up Teleport on a Ubuntu VM the other day and got it setup. All seems to work just fine via the browser etc, MFA and ssh into the server seems top notch all the time.
> 
> However, when I try to initiate a SOCKS proxy through dynamic port forwarding on my ubuntu linux client through TSH I get this all the time - randomly disconnects and causes Firefox to lose connection to the proxy server. Any thoughts?
> 
> $ **tsh ssh -D 127.0.0.1:9998 tech@ubuntuserver** tech@ubuntuserver:~$ logout the connection was closed on the remote side on 25 Jan 22 14:45 MST tech@tech-XPS-13-7390:~$ **tsh ssh -D 9998 tech@ubuntuserver** tech@ubuntuserver:~$ ERROR: wait: remote command exited without exit status or exit signal
> 
> tech@tech-XPS-13-7390:~$
> 
> I've tried it both ways for the command and both end up in this same error in a few short minutes.... (10)? Sometimes sooner, all random.
> 
> Thanks for any insight. I have port 443, 3023 to 3025, and 3080 forwarded to the Teleport box.

**Comment by webvictim** [comment]
I would check the logs on all Teleport servers (auth, proxy, SSH node) around the time that this disconnection occurs. There will likely be an error there which could be used to investigate further. At the moment there's not a lot of insight into why this might be happening.

**Comment by rusk0** [comment]
I am also facing a similar behavior, but when simply logging in with tsh ssh 

I am able to connect and unless i am idle no issues are noticed, however if i am idle for less than a minute i get disconnected with error :

Original Error: *ssh.ExitMissingError wait: remote command exited without exit status or exit signal


Debug error :

Original Error: *ssh.ExitMissingError wait: remote command exited without exit status or exit signal
Stack Trace:
	github.com/gravitational/teleport/lib/client/client.go:1628 github.com/gravitational/teleport/lib/client.(*NodeClient).RunInteractiveShell
	github.com/gravitational/teleport/lib/client/api.go:1823 github.com/gravitational/teleport/lib/client.(*TeleportClient).runShellOrCommandOnSingleNode
	github.com/gravitational/teleport/lib/client/api.go:1559 github.com/gravitational/teleport/lib/client.(*TeleportClient).SSH
	github.com/gravitational/teleport/tool/tsh/common/tsh.go:3229 github.com/gravitational/teleport/tool/tsh/common.onSSH.func1.1
	github.com/gravitational/teleport/lib/client/api.go:573 github.com/gravitational/teleport/lib/client.RetryWithRelogin
	github.com/gravitational/teleport/tool/tsh/common/tsh.go:3228 github.com/gravitational/teleport/tool/tsh/common.onSSH.func1
	github.com/gravitational/teleport/tool/tsh/common/tsh.go:3146 github.com/gravitational/teleport/tool/tsh/common.retryWithAccessRequest
	github.com/gravitational/teleport/tool/tsh/common/tsh.go:3227 github.com/gravitational/teleport/tool/tsh/common.onSSH
	github.com/gravitational/teleport/tool/tsh/common/tsh.go:1264 github.com/gravitational/teleport/tool/tsh/common.Run
	github.com/gravitational/teleport/tool/tsh/common/tsh.go:548 github.com/gravitational/teleport/tool/tsh/common.Main
	github.com/gravitational/teleport/tool/tsh/main.go:24 main.main
	runtime/proc.go:267 runtime.main
	runtime/asm_amd64.s:1650 runtime.goexit
User Message: wait: remote command exited without exit status or exit signal


I have tried various ssh client/server settings and some teleport.yaml settings on the node + some role settings on the teleport server ...

Nothing seems to address this issue .


some node syslog logs, that don't mean much to me :

Nov  3 10:00:30 $hostname teleport[2445250]: User Message: watcher is closed: error reading from server: EOF] cache/cache.go:972
Nov  3 10:00:30 $hostname teleport[2445250]: 2023-11-03T10:00:30Z WARN             Inventory control stream failed: inventory control stream closed: error reading from server: EOF inventory/inventory.go:196

**Comment by yair-segal** [comment]
We are also facing this issue. 
Teleport version 15.x.y
I'm not sure it's related, but - we are running Teleport cluster self-hosted, and when I reduced the proxy replica from 2 to 1, the issue still occurs, but less frequent.
